searchWithTermAndLocation

gkye opened the pull request:
added a function to accept a location parameter.

Commit:
searchWithTermAndLocation

add a function to accept a location parameter.

timothy1ee commented on Yelp/Business.swift:
```diff
@@ -91,4 +91,9 @@ class Business: NSObject {
+        YelpClient.sharedInstance.searchWithTermAndLocation(term, location: location, completion: completion)
+    }
+    
 }
```
Can you fix the whitespace here? There should be a space between functions, and there shouldn't be a space in the first line of the function.

timothy1ee commented on Yelp/YelpClient.swift:
```diff
@@ -57,10 +57,10 @@ class YelpClient: BDBOAuth1RequestOperationManager {
         var parameters: [String : AnyObject] = ["term": term, "ll": "37.785771,-122.406165"]
-
+        
         if sort != nil {
```
Can you remove this extra whitespace?

timothy1ee commented on Yelp/YelpClient.swift:
```diff
@@ -84,4 +84,26 @@ class YelpClient: BDBOAuth1RequestOperationManager {
+                completion(nil, error)
+        })!
+    }
 }
```
If you're adding a convenience method, can you chain everything through the same method? That way there is less duplicate code and less extraneous code paths where bugs can crop up.

Also, can you remove any print statements that you added for testing?

timothy1ee commented on Yelp/YelpClient.swift:
```diff
@@ -84,4 +84,26 @@ class YelpClient: BDBOAuth1RequestOperationManager {
+                
+        //change ll to location (ll return nil)
+        parameters["location"] = location;
+
```
Note that the reason we used "ll" was that causes the results to return a distance from the restaurant. I believe that adding "location" doesn't return a distance. I could be wrong there, but that's what I remember right now.

gkye replied:
You're correct, there's no distance return when the location parameters added.

timothy1ee commented:
Thanks for the pull request! I added some feedback on your changes.

Commit:
searchWithTermAndLocation

Location parameter added to main searchWithTerm function to allow
chaining with searchWithTermAndLocation. print statements removed and
extra whitespace removed.

gkye commented:
Thanks for the great feedback! I've made some changes accordingly.

timothy1ee commented:
Thanks, looks good. We have some students going through the homework using this repository right now. I'm going to test this a bit more before I merge.

Appreciate the pull request!

gkye commented:
No problem!
